app/goprotoc/goprotoc.go

```diff
@@ -286,11 +293,11 @@ func saveDescriptor(dest string, fds []*desc.FileDescriptor, includeImports, inc
 	for _, fd := range fds {
 		toFileDescriptorSet(fdsByName, &fdSet, fd, includeImports, includeSourceInfo)
 	}
-	if b, err := proto.Marshal(&fdSet); err != nil {
+	b, err := proto.Marshal(&fdSet)
+	if err != nil {
```

Comment:
This is actually a significant reason of why I have `golint` disabled in every project I have. I simply disagree very strongly with how draconian it wants to enforce this particular rule (and a few others). The `errcheck` one is less bad: it can at least be configured to ignore certain methods/patterns, which is nice.

In this particular case, this change is fine. But there are simply cases where typically accepted programming best practices -- to keep a variable's lifetime/scope as short as possible -- are flagged as lint violations. That's just terrible.

FWIW, I think even the authors of golint would say it's an anti-pattern to put it in your CI checks. This is from the official docs: https://github.com/golang/lint#purpose

> In short, this tool is not, and will never be, trustworthy enough for its suggestions to be enforced automatically, for example as part of a build process.

Reply:
I never knew that before - see my below comment. But totally fair.